pulp_container/tests/functional/api/test_recursive_remove.py

```diff
@@ -1,465 +1,488 @@
-)
+import pytest
+
 from pulp_container.tests.functional.constants import PULP_FIXTURE_1, REGISTRY_V2_REPO_PULP
 
-from pulpcore.client.pulp_container import (
-    ApiException,
-    ContainerContainerRemote,
-    ContainerContainerRepository,
-    ContainerRepositorySyncURL,
-    ContentManifestsApi,
-    ContentTagsApi,
-    DistributionsContainerApi,
-    PulpContainerNamespacesApi,
-    RemoveImage,
-    RemotesContainerApi,
-    RepositoriesContainerApi,
-    RepositoriesContainerPushApi,
-    RepositoriesContainerVersionsApi,
-    TagImage,
-)
-
-
-class TestRecursiveRemove(PulpTestCase):
+
+class TestRecursiveRemove:
     """
     Test recursively removing container content from a repository.
 
     This test targets the follow feature:
     https://pulp.plan.io/issues/5179
     """
 
-    @classmethod
-    def setUpClass(cls):
+    @pytest.fixture(scope="class")
+    def setup(
+        self,
+        container_bindings,
+        container_repository_factory,
+        container_remote_factory,
+        container_sync,
+    ):
         """Sync pulp/test-fixture-1 so we can copy content from it."""
-        api_client = gen_container_client()
-        cls.repositories_api = RepositoriesContainerApi(api_client)
-        cls.remotes_api = RemotesContainerApi(api_client)
-        cls.tags_api = ContentTagsApi(api_client)
-        cls.versions_api = RepositoriesContainerVersionsApi(api_client)
-
-        cls.from_repo = cls.repositories_api.create(ContainerContainerRepository(**gen_repo()))
-
-        remote_data = gen_container_remote(upstream_name=PULP_FIXTURE_1)
-        cls.remote = cls.remotes_api.create(ContainerContainerRemote(**remote_data))
-
-        sync_data = ContainerRepositorySyncURL(remote=cls.remote.pulp_href)
-        sync_response = cls.repositories_api.sync(cls.from_repo.pulp_href, sync_data)
-        monitor_task(sync_response.task)
-
-        cls.latest_from_version = cls.repositories_api.read(
-            cls.from_repo.pulp_href
-        ).latest_version_href
-
-    def setUp(self):
-        """Create an empty repository to copy into."""
-        self.to_repo = self.repositories_api.create(ContainerContainerRepository(**gen_repo()))
-        self.addCleanup(self.repositories_api.delete, self.to_repo.pulp_href)
-
-    @classmethod
-    def tearDownClass(cls):
-        """Delete things made in setUpClass. addCleanup feature does not work with setupClass."""
-        cls.repositories_api.delete(cls.from_repo.pulp_href)
-        cls.remotes_api.delete(cls.remote.pulp_href)
-        delete_orphans()
-
-    def test_repository_only_no_latest_version(self):
+        repo = container_repository_factory()
+        remote = container_remote_factory(upstream_name=PULP_FIXTURE_1)
+        container_sync(repo, remote)
+        from_repo = container_bindings.RepositoriesContainerApi.read(repo.pulp_href)
+        latest_from_version = from_repo.latest_version_href
+        return from_repo, latest_from_version
```

Comment:
Yet another copy of the `from_repo` fixture? maybe it should live in conftest...